### PR TITLE
Add support for multi-band raster ingest

### PR DESCRIPTION
**Pull request by sgillies**
Multi-band geospatial raster ingestion doesn't work. This PR corrects the code so that it does work.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+VERSION ?= 3.18.2
+
+dockertest-geospatial-amd64:
+	docker run --platform=linux/amd64 -it -v $(shell pwd):/app -v /tmp:/tmp --entrypoint=/bin/bash tiledbenterprise/notebook-python-r-julia-tilevcf-geo:$(VERSION) -c 'cd /app && python -m pip install --editable .[dev,tests] && python -m pytest $(OPTS)'
```
Comment (sgillies) [inline Makefile:4]: Added for local testing purposes. It's potentially useful for all maintainers; the notebook image is one way to get an environment with geospatial libs and packages that aren't on PyPI or conda-forge.

File: src/tiledb/cloud/geospatial/ingestion.py [hunk 2]
```diff
@@ -845,14 +845,13 @@ def ingest_raster_udf(
...
-                                    nodata=nodata,
-                                    indexes=[b + 1],
-                                )
-                                dst.write(chunk_arr, window=chunk_window)
+                            chunk_arr, _ = rasterio.merge.merge(
+                                input_datasets,
+                                bounds=chunk_bounds,
+                                nodata=nodata,
+                            )
+                            dst.write(chunk_arr, window=chunk_window)
```
Comment (sgillies) [inline src/tiledb/cloud/geospatial/ingestion.py:853]: `chunk_arr` as output of one method and input of the next helps ensure raster shape fidelity.

File: tests/test_geospatial.py [hunk 3]
```diff
@@ -289,6 +289,53 @@ def test_raster_ingest(self):
                 self.assertIsInstance(fltr, tiledb.ZstdFilter)
                 self.assertEqual(fltr.level, 7)
 
+    def test_rgb_raster_ingest(self):
```
Comment (sgillies) [inline tests/test_geospatial.py:292]: This test only asserts that ingest doesn't fail and that the output has the same shape as the input. That goes a long way, though. Rasterio already does a thorough job of testing raster merger.

Comment (sgillies) [reply tests/test_geospatial.py:298]: I've visually checked the result and it's correct.

**Comment by sgillies** [comment]
@normanb @nkules can you give this a look and review?

File: Makefile (hunk 1 above)
Comment (normanb) [reply Makefile:4]: 👍

File: src/tiledb/cloud/geospatial/ingestion.py (hunk 2 above)
Comment (normanb) [reply src/tiledb/cloud/geospatial/ingestion.py:853]: A reason for iterating over the bands was in the case of ingesting HSI data to reduce memory footprint. However the tile/cube sizes are relatively small, I was over thinking it as this is a generic ingestor.

**Review by normanb — APPROVED** [review]
👍

File: src/tiledb/cloud/geospatial/ingestion.py (hunk 2 above)
Comment (sgillies) [reply src/tiledb/cloud/geospatial/ingestion.py:853]: @normanb I'll restore that, then. I don't want to cause a regression.

File: src/tiledb/cloud/geospatial/ingestion.py [hunk 4]
```diff
@@ -845,14 +845,17 @@ def ingest_raster_udf(
                                 chunk_window, dst.transform
                             )
 
-                            for b in range(dst.count):
+                            for bidx in dst.indexes:
```
Comment (sgillies) [inline src/tiledb/cloud/geospatial/ingestion.py:848]: Improved rasterio usage here.

File: src/tiledb/cloud/geospatial/ingestion.py [hunk 5]
```diff
@@ -845,14 +845,17 @@ def ingest_raster_udf(
...
                                 chunk_arr, _ = rasterio.merge.merge(
                                     input_datasets,
                                     bounds=chunk_bounds,
                                     nodata=nodata,
-                                    indexes=[b + 1],
+                                    indexes=[bidx],
                                 )
-                                dst.write(chunk_arr, window=chunk_window)
+                                dst.write(
+                                    chunk_arr, indexes=[bidx], window=chunk_window
```
Comment (sgillies) [inline src/tiledb/cloud/geospatial/ingestion.py:856]: This is the minimal fix for the issue.

**Review by nkules — APPROVED** [review]

**Review by ihnorton — APPROVED** [review]

**Comment by ihnorton** [comment]
@sgillies please feel free to merge.

@JohnMoutafis let's get a release out with this change soon (cc @antalakas).

**Pull request merged**